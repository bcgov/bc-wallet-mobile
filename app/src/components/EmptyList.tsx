import { useTheme, testIdWithKey, Button, ButtonType } from 'aries-bifold'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Text, View, DeviceEventEmitter } from 'react-native'

import EmptyWallet from '../assets/img/emptyWallet.svg'
import { BCWalletEventTypes } from '../events/eventTypes'
export interface EmptyListProps {
  message?: string
}

const EmptyList: React.FC<EmptyListProps> = ({ message }) => {
  const { t } = useTranslation()
  const { ListItems } = useTheme()
  const [addCredentialPressed, setAddCredentialPressed] = useState<boolean>(false)

  useEffect(() => {
    const handle = DeviceEventEmitter.addListener(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED, (value?: boolean) => {
      const newVal = value === undefined ? !addCredentialPressed : value
      setAddCredentialPressed(newVal)
    })
    return () => {
      handle.remove()
    }
  }, [])

  const addCredentialPress = useCallback(() => {
    DeviceEventEmitter.emit(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED, !addCredentialPressed)
  }, [])

  return (
    <View style={{ marginTop: 100, height: '100%' }}>
      <EmptyWallet height={200} />
      <Text style={[ListItems.emptyList, { textAlign: 'center' }]} testID={testIdWithKey('NoneYet')}>
        {message || t('Global.NoneYet!')}
      </Text>
      <View style={{ margin: 25 }}>
        <Button
          title={t('Credentials.AddFirstCredential')}
          buttonType={ButtonType.Primary}
          onPress={addCredentialPress}
          disabled={addCredentialPressed}
        ></Button>
      </View>
    </View>
  )
}

export default EmptyList
