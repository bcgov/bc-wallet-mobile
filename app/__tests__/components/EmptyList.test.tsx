import { render } from '@testing-library/react-native'
import React from 'react'

import EmptyList from '../../src/components/EmptyList'

describe('EmptyList Component', () => {
  test('renders correctly', () => {
    const tree = render(<EmptyList />)
    expect(tree).toMatchSnapshot()
  })
})
