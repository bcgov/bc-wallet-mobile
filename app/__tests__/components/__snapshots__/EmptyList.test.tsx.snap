// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList Component renders correctly 1`] = `
<View
  style={
    {
      "height": "100%",
      "marginTop": 100,
    }
  }
>
  <
    height={200}
  />
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        {
          "textAlign": "center",
        },
      ]
    }
    testID="com.ariesbifold:id/NoneYet"
  >
    Global.NoneYet!
  </Text>
  <View
    style={
      {
        "margin": 25,
      }
    }
  >
    <View
      accessibilityLabel="Credentials.AddFirstCredential"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/AddFirstCredential"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
              false,
              false,
            ]
          }
        >
          Credentials.AddFirstCredential
        </Text>
      </View>
    </View>
  </View>
</View>
`;
